src/Entities/Player.cpp

```diff
@@ -2020,6 +2029,9 @@ bool cPlayer::DoMoveToWorld(cWorld * a_World, bool a_ShouldSendRespawn, Vector3d
 			{
 				ch->SendWeather(a_World->GetWeather());
 			}
+
+			// Update game mode
+			SetGameMode(a_World->GetGameMode());
```

Comment:
Perhaps this should be an option in the world.ini? Something like `ForceGameModeOnEnterWorld` or something.